Distinguishing entry "ident" from entry "name" in term notations. (#11841)
coq/coq: doc/sphinx/user-extensions/syntax-extensions.rst

     (at level 10, f global, a1, an at level 9).

In addition to ``global``, one can restrict the syntax of a
sub-expression by using the entry names ``ident``, ``name`` or ``pattern``
already seen in :ref:`NotationsWithBinders`, even when the
corresponding expression is not used as a binder in the right-hand
side. E.g.:

   .. todo: these two Set Warnings and the note should be removed when

>       ident is reactivated with its litteral meaning.

Reviewer: ```suggestion
      ident is reactivated with its literal meaning.
```

.. coqtop:: none

   Set Warnings "-deprecated-ident-entry".

.. coqtop:: in

   Notation "'apply_id' f a1 .. an" := (.. (f a1) .. an)
     (at level 10, f ident, a1, an at level 9).

.. coqtop:: none

   Set Warnings "+deprecated-ident-entry".

.. note:: As of version 8.13, the entry ``ident`` is a deprecated
          alias for ``name``. In the future, it is planned to strictly

>           parse an identifier (to the exclusion of :g:`_`).

Reviewer: ```suggestion
          parse an identifier (excluding :g:`_`).
```

custom entry ``expr`` be parsed or printed by the main grammar of term
up to the insertion of a pair of curly brackets.

Another special situation is when parsing global references or

> identifiers. To tell that a custom entry should parse identifiers, the
> following should be given:
> 

Reviewer: ```suggestion
identifiers. To indicate that a custom entry should parse identifiers,
use the following form:

```

.. coqtop:: in

   Notation "x" := x (in custom expr at level 0, x ident).

> Similarly, to tell that a custom entry should parse global references
> (i.e. qualified or non qualified identifiers), the following should be
> given:

Reviewer: ```suggestion
Similarly, to indicate that a custom entry should parse global references
(i.e. qualified or non qualified identifiers), use the following form:
```

      | only printing
      | format @string {? @string }
      explicit_subentry ::= ident

>       | name

Reviewer: The syntax blocks proceeded by `.. insertprodn` should be updated through doc_grammar rather than by hand.  If you don't do that, your change will disappear the next time doc_grammar updates the rsts.  Please run doc_grammar.

             due to legacy notation in the |Coq| standard library.
             It can be turned on with the ``-w disj-pattern-notation`` flag.

.. note::

   The explicit subentry ``ident`` is deprecated since version 8.12
   and should be replaced by ``name`` which is what it meant in
   practice (i.e. it parses either an identifier or the unnamed :g:`_`

>    variable). This does not apply to custom entries where it means as
>    expected an identifier.

Reviewer 2: ```suggestion
   variable). This does not apply to custom entries where it denotes
   an identifier, as expected.
```

             due to legacy notation in the Coq standard library.
             It can be turned on with the ``-w disj-pattern-notation`` flag.

.. note::

   As of version 8.13, the entry ``ident`` is a deprecated alias for

>    ``name``. In the future, it is planned to strictly parse an

Reviewer: ```suggestion
   ``name``. In the future, we plan to strictly parse an
```

Author: In general, I don't like the `we` so much as it would require to define who is "we" (which I would not be able to do).

Reviewer: It would mean the doc authors/Coq team.  In practice, I think you would make the change (and we, the rest of team, support that).  French doesn't use "we" this way?

We (there it is again!) try to avoid using the passive voice ("... it is planned") where possible.  The sentence is less readable with an unspecified subject, although in reference manuals it's often unavoidable.

Author: >  French doesn't use "we" this way?

French can use "we" (i.e. "nous") this way. The passive form would be fine here in French ("Il est prévu à terme que cette entrée corresponde strictement aux identifiers, à l'exclusion de _" or something like that).

In any case, I find the passive form a more informal promise than a "we" which would correspond to a "we" at the date of today and might not be the exact same "we" in one year from now on.

   identifier (to the exclusion of :g:`_`). If the intent was to use

>    ``ident`` in the sense of an identifier (to the exclusion of
>    :g:`_`), just silence the warning with

Reviewer: ```suggestion
   ``ident`` as an identifier (excluding :g:`_`), just silence the warning with
```

   :n:`Set Warnings "-deprecated-ident-entry"` and it should automatically
   get its intended meaning in version 8.15.

>    Similarly ``as ident`` is a deprecated alias for ``as name`` which
>    is planned to strictly parse an identifier in the future.  If the
>    intent was to use ``as ident`` in the sense of an identifier
>    (to the exclusion of :g:`_`), just silence the warning with

Reviewer: ```suggestion
   Similarly, ``as ident`` is a deprecated alias for ``as name``, which
   will only accept an identifier in the future.  If the
   intent was to use ``as ident`` as an identifier
   (excluding :g:`_`), just silence the warning with
```

   :n:`Set Warnings "-deprecated-as-ident-kind"`.

>    This deprecation does not apply to custom entries though where it
>    already denotes an identifier, as expected.

Reviewer: ```suggestion
   However, this deprecation does not apply to custom entries, where it
   already denotes an identifier, as expected.
```

Not sure it that's what you meant.
